crates/cast/bin/cmd/run.rs

```diff
@@ -161,6 +162,7 @@ impl RunArgs {
                     evm_version = Some(EvmVersion::Cancun);
                 }
             }
+            apply_chain_and_block_specific_env_changes::<AnyNetwork>(&mut env, block);
```

Comment:
I don't think we need this here, because this should already be applied when creating the env

https://github.com/foundry-rs/foundry/blob/4974a0891bf01511993f49138ebf81f4be207cdf/crates/evm/core/src/fork/init.rs#L83-L83

so we can leave this file unchanged

Reply:
Not exactly, see here:

https://github.com/foundry-rs/foundry/pull/9957/files/6769c1e937544ae46a4fdebe039737d8447e4010#diff-448a805ef6f60ed32e529c390289b7064010d85b3da89d32d3a6bd4766723769L152

Reply:
The block environment here is slightly different to a fork so that's why there are a few manual setup here (line 150-155). Emulating transactions need the state at block N but emulates at block N + 1 environment.

Reply:
ah I see